In [2]:
import numpy as np
import pandas as pd

import sys
sys.path.append("../")
from utils.helper_function import *

from rdkit import Chem
from rdkit.Chem import MACCSkeys, rdFingerprintGenerator
from rdkit import DataStructs

import warnings

warnings.filterwarnings("ignore", category=DeprecationWarning)
warnings.filterwarnings("ignore", category=UserWarning)

In [3]:
xls = pd.ExcelFile('../data/data_sixpk.xlsx')
df = pd.read_excel(xls, 'AcuteInhal')
df2 = pd.read_excel(xls,'SkinIrritation')


In [4]:
df.head()

CASRN                       SMILES  Activity
0  100-00-5  [O-][N+](=O)C1=CC=C(Cl)C=C1         1
1  100-20-9  ClC(=O)C1=CC=C(C=C1)C(Cl)=O         1
2  100-37-8                   CCN(CC)CCO         1
3  100-43-6               C=CC1=CC=NC=C1         1
4  100-44-7               ClCC1=CC=CC=C1         1

In [5]:
df2.head()

CASRN                       SMILES  Activity
0  100-02-7   OC1=CC=C(C=C1)[N+]([O-])=O         1
1  100-18-5      CC(C)C1=CC=C(C=C1)C(C)C         1
2  100-20-9  ClC(=O)C1=CC=C(C=C1)C(Cl)=O         1
3  100-36-7                   CCN(CC)CCN         1
4  100-37-8                   CCN(CC)CCO         1

In [6]:
smiles = df['SMILES']
activity = df['Activity'].to_numpy()

smiles2 = df2['SMILES']
activity2 = df2['Activity'].to_numpy()

##### MODI Index for MACCS structural key

In [7]:
fp_maccs = maccs_fp(smiles= smiles)
modiIdx_maccs = getMODIindex(df= df, fps= fp_maccs)

print(f"MODI Index using MACCS structural key technique for Acute Inhalation: {modiIdx_maccs}")

MODI Index using MACCS structural key technique for Acute Inhalation: 0.6791455886135642


In [14]:
fp_maccs2 = maccs_fp(smiles= smiles2)
modiIdx_maccs2 = getMODIindex(df= df2, fps= fp_maccs2)

print(f"MODI Index using MACCS structural key technique for Skin Irritation: {modiIdx_maccs2}")

MODI Index using MACCS structural key for Skin Irritation: 0.6791455886135642


##### MODI Index for Morgen Fringerprints

In [8]:
fp_morgen = morgen_fp(smiles= smiles)
modiIdx_morgen = getMODIindex(df= df, fps= fp_morgen)

print(f"MODI Index using the Morgen Fringerprint approach for Acute Inhalation: {modiIdx_morgen}")

MODI Index using the Morgen Fringerprint approach for Acute Inhalation: 0.6444203892651976


In [9]:
fp_morgen2 = morgen_fp(smiles= smiles2)
modiIdx_morgen2 = getMODIindex(df= df2, fps= fp_morgen2)

print(f"MODI Index using the Morgen Fringerprint approach for Skin Irritation: {modiIdx_morgen2}")

MODI Index using the Morgen Fringerprint approach for Skin Irritation: 0.6444203892651976


##### MODI Index for Mordred Fringerprint

In [ ]:
fp_mordred = mordred_fp(smiles= smiles)
modiIdx_mordred = getMODIindex(df= df, fps= fp_mordred, fp= 'customT')

print(f"MODI Index for Mordred Fringerprint: {modiIdx_mordred}")

In [ ]:
print(fp_mordred)

In [ ]:
calc = Calculator(descriptors, ignore_3D=True)
molecules = [Chem.MolFromSmiles(smi) for smi in smiles]
descriptors = calc.pandas(molecules)
print(descriptors)

In [10]:
descriptors.fillna(0, inplace=True)


In [ ]:
print(descriptors)

In [14]:
getMODIindex(df= df, fps= descriptors.values.tolist() ,fp='bal')

TypeError: can't multiply sequence by non-int of type 'list'

## MACCS Structural key 
> MODI Index using MACCS structural key technique for Acute Inhalation: 0.6791455886135642
     
> MODI Index using MACCS structural key for Skin Irritation: 0.6791455886135642
    

## Morgen FingerPrint
> MODI Index using the Morgen Fringerprint approach for Acute Inhalation: 0.6444203892651976  
        
> MODI Index using the Morgen Fringerprint approach for Skin Irritation: 0.6444203892651976     

## Mordred Fingerprint
> MODI Index using the Mordred Fringerprint approach for Acute Inhalation:  
       
> MODI Index using the Mordred Fringerprint approach for Skin Irritation:      

# =====================================================================

### Acute  Oral 

In [11]:
xls = pd.ExcelFile('../data/data_sixpk.xlsx')
df3 = pd.read_excel(xls, 'AcuteOral')
df3.head()

CASRN                       SMILES  Activity
0  100-00-5  [O-][N+](=O)C1C=CC(Cl)=CC=1         1
1  100-01-6   NC1=CC=C(C=C1)[N+]([O-])=O         1
2  100-02-7   [O-][N+](=O)C1C=CC(O)=CC=1         1
3  100-03-8        OS(=O)C1C=CC(Cl)=CC=1         1
4  100-06-1        CC(=O)C1=CC=C(C=C1)OC         1

In [13]:
smiles_ao = df3['SMILES']
activity_ao = df3['Activity'].to_numpy()

MACCS 

In [14]:
fp_maccs_ao = maccs_fp(smiles= smiles_ao)
modiIdx_maccs_ao = getMODIindex(df= df3, fps= fp_maccs_ao)

print(f"MODI Index using MACCS structural key technique for Acute Inhalation: {modiIdx_maccs_ao}")

KeyboardInterrupt: 

Morgen FringerPrint 

In [ ]:
fp_morgen_ao = morgen_fp(smiles= smiles_ao)
modiIdx_morgen_ao = getMODIindex(df= df3, fps= fp_morgen_ao)

print(f"MODI Index using the Morgen Fringerprint approach for Acute Inhalation: {modiIdx_morgen_ao}")

Mordred FingerPrint